### PR TITLE
cloud/router: fix and optimize rule code

### PR DESCRIPTION
**Pull request by Iceber**
Signed-off-by: Iceber Gu <wei.cai-nat@daocloud.io>

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:
fix print message content type and optimize code

Abstracting out some functions:
1. `handleRuleEndpoint` and `handleRule` is used to add listener to `MessageHandlerInstance`
2. `getSourceOfRule` and `getTargetOfRule` is used to get source and target of the rule.

optimize statushandler.go's code, abstract out the `handleStatusAndReset`

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cloud/pkg/router/rule/rule.go [hunk 1]
```diff
@@ -14,67 +14,96 @@ import (
...
 	registerListener()
 }
 
 func registerListener() {
-	eKey := fmt.Sprintf("%s/%s", modules.EdgeControllerModuleName, model.ResourceTypeRuleEndpoint)
-	listener.MessageHandlerInstance.AddListener(eKey, func(data interface{}) (interface{}, error) {
-		message := data.(*model.Message)
-		ruleEndpoint, ok := message.GetContent().(*routerv1.RuleEndpoint)
-		if !ok {
-			klog.Warningf("object type: %T unsupported", ruleEndpoint)
```
Comment (Iceber) [inline cloud/pkg/router/rule/rule.go:30]: fix: `klog.Warningf("object type: %T unsupported", message.Content)`

File: cloud/pkg/router/rule/rule.go [hunk 2]
```diff
@@ -128,33 +161,36 @@ func addRule(rule *routerv1.Rule) error {
 		return resp, nil
 	})
 	if err != nil {
-		klog.Errorf("add rule %s:%s failed, err: %v", rule.Name, rule.Name, err)
```
Comment (Iceber) [inline cloud/pkg/router/rule/rule.go:131]: fix: rule.Namespace/rule.Name

File: cloud/pkg/router/rule/statushandler.go [hunk 3]
```diff
@@ -18,86 +18,78 @@ type ErrorMsg struct {
 	Timestamp time.Time
 }
 
-var ResultChannel chan ExecResult
-var StopChan chan bool
```
Comment (Iceber) [inline cloud/pkg/router/rule/statushandler.go:22]: Not used anywhere

**Comment by fisherxu** [comment]
cc @liufen90

**Review by WintonChan — APPROVED** [review]

**Pull request merged**